sinopia_client/docs/LDPApi.md

```diff
@@ -133,6 +136,7 @@ Name | Type | Description  | Notes
  **groupID** | **String**| The group who is defining it&#39;s own resources or graph within Sinopia. | 
  **resource** | [**Resource**](Resource.md)| Resource to insert into container | 
  **slug** | **String**| The suggested URI path for the resource. | [optional] 
+ **link** | **String**| specifies container type. | [optional] 
```

Comment:
If left out, will Trellis assign a resource the LDP type we want? That's cool, if so!

Reply:
ah, no, unfortunately we have to specify it manually:
https://github.com/LD4P/sinopia_server/pull/66/files#diff-9c98b1f96912899c02b40c5bb91a07dcR108

in a couple other parts of the code i manually implemented a default value, e.g.:
https://github.com/LD4P/sinopia_server/pull/66/files#diff-7ef00ea8d37b22fc3b6ea30d5a63166eR84

i can't remember what trellis does for the type if you don't specify one, but it wasn't what we wanted in at least the basic container case, and it seems like we always want it specified as something.  additionally, while swagger provides a field for specifying a default param value, my experiments with the JS codegen tool show that it's not something that's implemented (specifying made no difference in generated code).

in the usage pointed out in the first link, i was hesitant to make `ldp#NonRDFSource` the default value because i figured the predominant use case would be regular `ldp#RDFSource`.  but i was hesitant to code that as a default value similar to the second link, because i hadn't yet written a test for regular resource creation (which was the next thing i figured i'd do, along with which i'd code the default value thing for `ldp#RDFSource`).

comments about providing default values and modifying auto-gen code w/ said auto-gen code...

Reply:
it says optional -- is there a way to indicate it's required?